code/modules/admin/admin_verbs.dm

```diff
@@ -358,7 +360,10 @@ GLOBAL_LIST_INIT(view_runtimes_verbs, list(
 	set category = "Admin"
 	set name = "Aghost"
 
-	if(!check_rights(R_ADMIN|R_MOD))
+	if(!check_rights(R_ADMIN|R_MOD))  // todo verify this logic checks out
+		return
+
+	if(!check_rights(R_ADMIN|R_MOD) && check_rights(R_MENTOR) && !HAS_MIND_TRAIT(mob, TRAIT_MOBSERVE))
```

Comment:
I don't think you meant to leave this in, since it's changing Aghost.

Reply:
Yes, this is intended. Admin ghost will not be given to mentors as a verb, but it can be called by the code for mentor observing. 

Note the trait here: MOBSERVE will only be given when a mentor has started observing.

Reply:
It'd be better to extract the code the MOBSERVE uses from Aghost into a proc, and call that instead, rather than directly calling a verb that mentors don't have. Even if that's the entire thing, making Aghost a thin wrapper around the proc would still be better than calling Aghost directly.

Reply:
That's fair! I mostly just didn't want to duplicate the code, so I can def move it out into a wrapper

Reply:
Addressed